### PR TITLE
MODSuit clothing icon files can be overridden by its skin

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
This allows the MODSuit skin to change the icon and worn icon files of the modsuit pieces.

## Why It's Good For The Game
It's good for downstreams.

## Changelog

Nothing player-facing

### CODE REVIEW

File: code/modules/mod/mod_control.dm [hunk 1]
```diff
@@ -135,6 +135,8 @@
 		piece.permeability_coefficient = theme.permeability_coefficient
 		piece.siemens_coefficient = theme.siemens_coefficient
 		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
+		piece.icon = theme.icon_override
+		piece.worn_icon = theme.worn_icon_override
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:139]: this is run only on initialize, put it on set_mod_skin too

File: code/modules/mod/mod_theme.dm [hunk 2]
```diff
@@ -49,6 +49,10 @@
 	var/list/inbuilt_modules = list()
 	/// Modules blacklisted from the MOD.
 	var/list/module_blacklist = list()
+	/// Icon file override for if the mod clothing uses another icon file.
+	var/icon_override = 'icons/obj/clothing/modsuit/mod_clothing.dmi'
+	/// Worn Icon file override for if the mod clothing uses another worn icon file.
+	var/worn_icon_override = 'icons/mob/clothing/mod.dmi'
```
Comment (Fikou) [inline code/modules/mod/mod_theme.dm:53]: this shouldnt be on theme, it should be on skins

File: code/modules/mod/mod_control.dm (hunk 1 above)
Comment (Zonespace27) [reply code/modules/mod/mod_control.dm]: Resolved

File: code/modules/mod/mod_theme.dm (hunk 2 above)
Comment (Zonespace27) [reply code/modules/mod/mod_theme.dm]: Resolved

File: code/modules/mod/mod_control.dm [hunk 3]
```diff
@@ -135,6 +136,10 @@
 		piece.permeability_coefficient = theme.permeability_coefficient
 		piece.siemens_coefficient = theme.siemens_coefficient
 		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
+		if(selected_skin[MOD_ICON_OVERRIDE])
+			piece.icon = selected_skin[MOD_ICON_OVERRIDE]
+		if(selected_skin[MOD_WORN_ICON_OVERRIDE])
+			piece.worn_icon = selected_skin[MOD_WORN_ICON_OVERRIDE]
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:142]: maybe we can just call set_mod_skin here

Comment (Zonespace27) [reply code/modules/mod/mod_control.dm]: Done

File: code/modules/mod/mod_control.dm [hunk 4]
```diff
@@ -134,8 +134,7 @@
 		piece.min_cold_protection_temperature = theme.min_cold_protection_temperature
 		piece.permeability_coefficient = theme.permeability_coefficient
 		piece.siemens_coefficient = theme.siemens_coefficient
-		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
-	update_flags()
+	set_mod_skin(skin)
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:137]:
```suggestion
	set_mod_skin(new_skin || theme.default_skin)
```
do this and remove line 103

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**